Cannot built Seastar with DPDK on Ubuntu 22.04

Build fails with this error:
```
ninja -C build/debug
ninja: Entering directory `build/debug'
[20/263] Linking CXX shared library libseastar.so
FAILED: libseastar.so 
: && /usr/bin/g++ -fPIC -g   -shared -Wl,-soname,libseastar.so -o libseastar.so seastar-dpdk.o CMakeFiles/seastar.dir/src/core/alien.cc.o CMakeFiles/seastar.dir/src/core/file.cc.o CMakeFiles/seastar.dir/src/core/fair_queue.cc.o CMakeFiles/seastar.dir/src/core/reactor_backend.cc.o CMakeFiles/seastar.dir/src/core/thread_pool.cc.o CMakeFiles/seastar.dir/src/core/app-template.cc.o CMakeFiles/seastar.dir/src/core/dpdk_rte.cc.o CMakeFiles/seastar.dir/src/core/exception_hacks.cc.o CMakeFiles/seastar.dir/src/core/execution_stage.cc.o CMakeFiles/seastar.dir/src/core/fsnotify.cc.o CMakeFiles/seastar.dir/src/core/fsqual.cc.o CMakeFiles/seastar.dir/src/core/fstream.cc.o CMakeFiles/seastar.dir/src/core/future.cc.o CMakeFiles/seastar.dir/src/core/future-util.cc.o CMakeFiles/seastar.dir/src/core/linux-aio.cc.o CMakeFiles/seastar.dir/src/core/memory.cc.o CMakeFiles/seastar.dir/src/core/metrics.cc.o CMakeFiles/seastar.dir/src/core/on_internal_error.cc.o CMakeFiles/seastar.dir/src/core/posix.cc.o CMakeFiles/seastar.dir/src/core/prometheus.cc.o CMakeFiles/seastar.dir/src/core/program_options.cc.o CMakeFiles/seastar.dir/src/core/reactor.cc.o CMakeFiles/seastar.dir/src/core/resource.cc.o CMakeFiles/seastar.dir/src/core/sharded.cc.o CMakeFiles/seastar.dir/src/core/scollectd.cc.o CMakeFiles/seastar.dir/src/core/systemwide_memory_barrier.cc.o CMakeFiles/seastar.dir/src/core/smp.cc.o CMakeFiles/seastar.dir/src/core/sstring.cc.o CMakeFiles/seastar.dir/src/core/thread.cc.o CMakeFiles/seastar.dir/src/core/uname.cc.o CMakeFiles/seastar.dir/src/core/io_queue.cc.o CMakeFiles/seastar.dir/src/core/semaphore.cc.o CMakeFiles/seastar.dir/src/core/condition-variable.cc.o CMakeFiles/seastar.dir/src/http/api_docs.cc.o CMakeFiles/seastar.dir/src/http/common.cc.o CMakeFiles/seastar.dir/src/http/file_handler.cc.o CMakeFiles/seastar.dir/src/http/httpd.cc.o CMakeFiles/seastar.dir/src/http/json_path.cc.o CMakeFiles/seastar.dir/src/http/matcher.cc.o CMakeFiles/seastar.dir/src/http/mime_types.cc.o CMakeFiles/seastar.dir/src/http/reply.cc.o CMakeFiles/seastar.dir/src/http/routes.cc.o CMakeFiles/seastar.dir/src/http/transformers.cc.o CMakeFiles/seastar.dir/src/http/url.cc.o CMakeFiles/seastar.dir/src/http/client.cc.o CMakeFiles/seastar.dir/src/http/request.cc.o CMakeFiles/seastar.dir/src/json/formatter.cc.o CMakeFiles/seastar.dir/src/json/json_elements.cc.o CMakeFiles/seastar.dir/src/net/arp.cc.o CMakeFiles/seastar.dir/src/net/config.cc.o CMakeFiles/seastar.dir/src/net/dhcp.cc.o CMakeFiles/seastar.dir/src/net/dns.cc.o CMakeFiles/seastar.dir/src/net/dpdk.cc.o CMakeFiles/seastar.dir/src/net/ethernet.cc.o CMakeFiles/seastar.dir/src/net/inet_address.cc.o CMakeFiles/seastar.dir/src/net/ip.cc.o CMakeFiles/seastar.dir/src/net/ip_checksum.cc.o CMakeFiles/seastar.dir/src/net/native-stack.cc.o CMakeFiles/seastar.dir/src/net/net.cc.o CMakeFiles/seastar.dir/src/net/packet.cc.o CMakeFiles/seastar.dir/src/net/posix-stack.cc.o CMakeFiles/seastar.dir/src/net/proxy.cc.o CMakeFiles/seastar.dir/src/net/socket_address.cc.o CMakeFiles/seastar.dir/src/net/stack.cc.o CMakeFiles/seastar.dir/src/net/tcp.cc.o CMakeFiles/seastar.dir/src/net/tls.cc.o CMakeFiles/seastar.dir/src/net/udp.cc.o CMakeFiles/seastar.dir/src/net/unix_address.cc.o CMakeFiles/seastar.dir/src/net/virtio.cc.o CMakeFiles/seastar.dir/src/rpc/lz4_compressor.cc.o CMakeFiles/seastar.dir/src/rpc/lz4_fragmented_compressor.cc.o CMakeFiles/seastar.dir/src/rpc/rpc.cc.o CMakeFiles/seastar.dir/src/util/alloc_failure_injector.cc.o CMakeFiles/seastar.dir/src/util/backtrace.cc.o CMakeFiles/seastar.dir/src/util/conversions.cc.o CMakeFiles/seastar.dir/src/util/exceptions.cc.o CMakeFiles/seastar.dir/src/util/file.cc.o CMakeFiles/seastar.dir/src/util/log.cc.o CMakeFiles/seastar.dir/src/util/process.cc.o CMakeFiles/seastar.dir/src/util/program-options.cc.o CMakeFiles/seastar.dir/src/util/read_first_line.cc.o CMakeFiles/seastar.dir/src/util/tmp_file.cc.o CMakeFiles/seastar.dir/src/util/short_streams.cc.o CMakeFiles/seastar.dir/src/websocket/server.cc.o  /usr/lib/x86_64-linux-gnu/libboost_program_options.so  /usr/lib/x86_64-linux-gnu/libboost_thread.so  /usr/lib/x86_64-linux-gnu/libcares.so  /usr/lib/x86_64-linux-gnu/libcryptopp.so  /usr/lib/x86_64-linux-gnu/libfmt.so.8.1.1  /usr/lib/x86_64-linux-gnu/liblz4.so  -ldl  /usr/lib/x86_64-linux-gnu/libgnutls.so  /usr/lib/x86_64-linux-gnu/libsctp.so  /usr/lib/x86_64-linux-gnu/libyaml-cpp.so  /usr/lib/x86_64-linux-gnu/libhwloc.so  /usr/lib/x86_64-linux-gnu/liburing.so  /usr/lib/x86_64-linux-gnu/libnuma.so  /usr/lib/x86_64-linux-gnu/libboost_chrono.so  /usr/lib/x86_64-linux-gnu/libboost_date_time.so  /usr/lib/x86_64-linux-gnu/libboost_atomic.so  -Wl,--as-needed  -latomic  -fsanitize=address  -fsanitize=undefined  -fno-sanitize=vptr  _cooking/installed/lib/librte_cfgfile.a  _cooking/installed/lib/librte_cmdline.a  _cooking/installed/lib/librte_ethdev.a  _cooking/installed/lib/librte_hash.a  _cooking/installed/lib/librte_mbuf.a  _cooking/installed/lib/librte_eal.a  _cooking/installed/lib/librte_kvargs.a  _cooking/installed/lib/librte_mempool.a  _cooking/installed/lib/librte_mempool_ring.a  _cooking/installed/lib/librte_pmd_bnxt.a  _cooking/installed/lib/librte_pmd_cxgbe.a  _cooking/installed/lib/librte_pmd_e1000.a  _cooking/installed/lib/librte_pmd_ena.a  _cooking/installed/lib/librte_pmd_enic.a  _cooking/installed/lib/librte_pmd_qede.a  _cooking/installed/lib/librte_pmd_i40e.a  _cooking/installed/lib/librte_pmd_ixgbe.a  _cooking/installed/lib/librte_pmd_nfp.a  _cooking/installed/lib/librte_pmd_ring.a  _cooking/installed/lib/librte_pmd_vmxnet3_uio.a  _cooking/installed/lib/librte_ring.a  _cooking/installed/lib/librte_net.a  _cooking/installed/lib/librte_timer.a  _cooking/installed/lib/librte_pci.a  _cooking/installed/lib/librte_bus_pci.a  _cooking/installed/lib/librte_bus_vdev.a  _cooking/installed/lib/librte_pmd_fm10k.a  _cooking/installed/lib/librte_pmd_sfc_efx.a && :
/usr/bin/ld: seastar-dpdk.o: relocation R_X86_64_TPOFF32 against symbol `per_lcore__lcore_id' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
[22/263] Building CXX object CMakeFiles/seastar_perf_testing.dir/tests/perf/perf_tests.cc.o

```

--- a/cooking_recipe.cmake
+++ b/cooking_recipe.cmake
@@ -282,6 +282,10 @@ endif()
 
 # gcc 10 defaults to -fno-common, which dpdk is not prepared for
 set (dpdk_extra_cflags "-Wno-error -fcommon -fpie")
+if (BUILD_SHARED_LIBS)
+  string (APPEND dpdk_extra_cflags " -fPIC")
+endif ()
+
 set (dpdk_args
   "EXTRA_CFLAGS=${dpdk_extra_cflags}"
   O=<BINARY_DIR>

--- a/cooking_recipe.cmake
+++ b/cooking_recipe.cmake
@@ -280,9 +280,10 @@ else()
   set (dpdk_quadruple ${CMAKE_SYSTEM_PROCESSOR}-native-linuxapp-gcc)
 endif()
 
+# gcc 10 defaults to -fno-common, which dpdk is not prepared for
+set (dpdk_extra_cflags "-Wno-error -fcommon -fpie")
 set (dpdk_args
-  # gcc 10 defaults to -fno-common, which dpdk is not prepared for
-  "EXTRA_CFLAGS=-Wno-error -fcommon -fpie"
+  "EXTRA_CFLAGS=${dpdk_extra_cflags}"
   O=<BINARY_DIR>
   DESTDIR=<INSTALL_DIR>
   T=${dpdk_quadruple})

